lib/tom_queue/deferred_work_manager.rb

```diff
@@ -113,6 +121,24 @@ def stop
       deferred_set && deferred_set.interrupt
     end
 
-  end
+    # Register handlers to shut down the manager gracefully
+    def register_signal_handlers
+      Thread.main[:signals] = []
+
+      [:QUIT, :INT, :TERM].each do |sig|
+        trap(sig) do
+          Thread.main[:signals] << sig
+        end
+      end
```

Comment:
Will this cause deadlock in some edge cases (e.g. when DeferredWorkSet#pop is called at the same time), since `interrupt` is using `mutex` https://bugs.ruby-lang.org/issues/7917